File: CiF071BroadcastRecievers/app/src/main/java/com/gmail/expencive/cif071broadcastrecievers/MainActivity.java
package com.gmail.expencive.cif071broadcastrecievers;

import android.content.IntentFilter;
import android.net.ConnectivityManager;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;

public class MainActivity extends AppCompatActivity {

    OrderedReceiver1 orderedReceiver1 = new OrderedReceiver1();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        IntentFilter filter = new IntentFilter("com.gmail.expencive.EXAMPLE_ACTION");
        filter.setPriority(0);
        registerReceiver(orderedReceiver1, filter);


    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        unregisterReceiver(orderedReceiver1);
    }
}

File: KotlinSingletonExample/app/src/main/java/vk/expencive/kotlinsingletonexample/MainActivity.kt
package vk.expencive.kotlinsingletonexample

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.os.UserHandle
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProvider
import vk.expencive.kotlinsingletonexample.models.User

class MainActivity : AppCompatActivity() {

    private lateinit var viewModel: MainViewModel

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        println("DEBUG: ${ExampleSingleton.singletonUser}")
        println("DEBUG: ${ExampleSingleton.singletonUser.hashCode()}")

        viewModel = ViewModelProvider(this).get(MainViewModel::class.java)

        viewModel.user.observe(this, Observer {
            println("DEBUG: $it")
        })

        viewModel.setUserId("1")

        println("DEBUG: ExampleSingleton: ${ExampleSingleton}")
    }


    override fun onDestroy() {
        super.onDestroy()
        viewModel.cancelJobs()
    }
}
